protos/sigstore_verification.proto

```diff
@@ -54,6 +54,10 @@ message ArtifactVerificationOptions {
                 bool perform_online_verification = 2;
                 // Disable verification for transparency logs.
                 bool disable = 3;
+                // Verify SET timestamps indicates that the timestamp from
+                // the SET should be used when verifying the X.509
+                // certifiacte chain
+                bool verify_set_timestamp = 4;
```

Comment:
My last bullet point is wrong. I think `verify_set` must be true in order to count towards the ObserverTimestampOptions. If `verify_set=true` is set and only the TSA threshold is set up, then no threshold would be met (working as intended).

(tldr below is just me trying this out in a client)

As I've been prototyping this in sigstore-go, the only question I have is if it's sufficient to only set `WithTransparencyLog`. This means that the threshold for number of logs and timestamps must match, which I would prefer each should be explicitly configured.

I think the correct client approach would be to require when is used `WithTransparencyLog(threshold int, verifySET bool)`, also one of the following is required: `WithTSA(threshold)` or `WithObserverTimestamp(threshold)` or `WithInsecureNoTimestamp`. If `verifySET==true`, then the timestamps returned from verifying the log can count towards the `WithObserverTimestamp` threshold. If `WithTransparencyLog(1, true)` and `WithTSA(1)`, then SETs could be used to satisfy the log threshold, but not the timestamp threshold.

Reply:
OK, sorry I keep going back and forth, but I thought of one case this doesn't handle. How would you specify requiring X log proofs, 0 promises, and Y timestamps that are either SETs or TSA timestamps? This was the point of my last bullet point, some way to specify SETs shouldn't count towards.

So either:
* we need SETs to be their own struct to support this, or:
* a comment that `verify_set` only dictates whether or not SETs count towards the log threshold. We will always still try to verify SETs and extract timestamps, whose policy is dictated by ObserverTimestampOptions.

cc @steiza @phillmv 

Sorry for rambling on about this, hopefully this makes sense! Late Friday isn't the time to figure this out 😄

Reply:
As v0.2 has the SET as optional, and the proof as required. My thinking in this discussion was only around how the SET counts toward the timestamp threshold, not against the tlog threshold, and so:

> a comment that verify_set only dictates whether or not SETs count towards the log threshold. We will always still try to verify SETs and extract timestamps, whose policy is dictated by ObserverTimestampOptions.

would be sort of the opposite: _a comment that verify_set only dictates that it count towards the timestamp_ (if present)

This would mean that if the tlog does only contain a proof, this entry does not count up towards the timestamp threshold. I.e a tlog entry without SET but verified with `verify_set=true` would not fail, just not increase the counter towards timestamp threshold.

> How would you specify requiring X log proofs, 0 promises, and Y timestamps that are either SETs or TSA timestamps

Like this: 
WithTLog(x, true) + WithObserverTimestamp(y)

My reading:
WithTLog(x, true): We require x inclsuion proofs (required by v0.2 bundle), `verifySET=true` we trust the timestamp from the SET if included.
WithObserverTimestamp(y): we require at least y timestamps (as `verifySET=true`, the SET counts if present, if not then y TSA timestamps is required).

Reply:
Good point about v0.2 requiring inclusion proof. Continuing discussion on sigstore-go.